app.rb

```diff
@@ -0,0 +1,75 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require './config/data_mapper'
+require 'pry'
+
+
+class ChitterApp < Sinatra::Base
+  enable :sessions
+
+  get '/' do
+    @messages = Message.all
+    current_user
+    if logged_in?
+      erb :logged_in
+    else
+      erb :index
+    end
+  end
+
+  post '/post_msg' do
+    Message.create(
+      peep: params[:msg],
+      username: current_username
+    )
+    redirect '/'
+  end
+
+  get '/signup' do
+    erb :signup
+  end
+
+  post '/signup' do
+    user = User.create(
+      username: params[:username],
+      mail: params[:mail],
+      password: params[:password]
+    )
+    session[:userid] = user.id if user
+    redirect '/'
+  end
+
+  get '/login' do
+    erb :login
+  end
+
+  post '/login' do
+    user = User.auth(params[:username], params[:password])
+    session[:userid] = user.id if user
+    redirect '/'
+  end
+
+  get '/logout' do
```

Comment:
Maybe it can be an other approach to do a delete '/sessions' route for the log out, which gonna "kill" the session of the user.
also, you can redirect your user to the homepage instead of having a view for log out.